storageservice/modules/blobs/endpoint.bal

```diff
@@ -687,23 +687,29 @@ public isolated client class BlobClient {
         int i = 0; // Index of current block
         int remainingBytes = fileSize; // Remaining bytes to upload
         string[] blockIdArray = []; // List of blockIds
+        boolean isOver = false;
 
-        var fileStream = check io:fileReadBlocksAsStream(filePath, MAX_BLOB_UPLOAD_SIZE);
-        error? e = fileStream.forEach(function(io:Block byteBlock) {
-            string blockId = blobName + COLON_SYMBOL + i.toString();
-            blockIdArray[i] = blockId;
-                    
-            if (remainingBytes < MAX_BLOB_UPLOAD_SIZE) {
-                byte[] lastByteArray = byteBlock.slice(0, remainingBytes);
-                _ = checkpanic self->putBlock(containerName, blobName, blockId, lastByteArray);
-                log:printInfo("Upload successful");
+        stream<io:Block, io:Error?> fileStream = check io:fileReadBlocksAsStream(filePath, MAX_BLOB_UPLOAD_SIZE);
+        while !isOver {
+            record {| byte[] & readonly value; |}? byteBlock  = check fileStream.next();
+            if(byteBlock is ()) {
+                isOver = true;
             } else {
-                _ = checkpanic self->putBlock(containerName, blobName, blockId, byteBlock);
-                remainingBytes -= MAX_BLOB_UPLOAD_SIZE;
-                log:printInfo("Remaining bytes to upload: " + remainingBytes.toString() + "Bytes");
-                i += 1;  
-            }             
-        });
+                string blockId = blobName + COLON_SYMBOL + i.toString();
+                blockIdArray[i] = blockId;
+                        
+                if (remainingBytes < MAX_BLOB_UPLOAD_SIZE) {
+                    byte[] lastByteArray = byteBlock.value.slice(0, remainingBytes);
+                    _ = check self->putBlock(containerName, blobName, blockId, lastByteArray);
+                    log:printInfo("Upload successful");
+                } else {
+                    _ = check self->putBlock(containerName, blobName, blockId, byteBlock.value);
+                    remainingBytes -= MAX_BLOB_UPLOAD_SIZE;
+                    log:printInfo("Remaining bytes to upload: " + remainingBytes.toString() + "Bytes");
+                    i += 1;  
```

Comment:
Shall we say `i = i + 1` for readability. That's our general practice in the team.